booking.py

```diff
@@ -188,25 +188,27 @@ def add_tagged_friends(friends):
                         f2.tag = t
         return friends
 
+    #files_base_directory: ntuple/era
+    #friends_base_directory: friends/friend_type/era
     root_files = []
     for f in file_names:
-        for g in os.listdir(os.path.join(files_base_directory, f)):
-            print(os.path.join(files_base_directory, f, g))
-            root_files.append(os.path.join(files_base_directory, f, g))
+        for g in os.listdir(os.path.join(files_base_directory, f, channel)):
+            root_files.append((os.path.join(files_base_directory, f, channel, g),f))
     ntuples = []
-    for root_file, file_name in zip(root_files, file_names):
```

Comment:
Is there a reason to combine this already in the loop above and not use the zipped version?

Reply:
i impemented it this way, because, in crown we have many files which correspond to the same "file name" (process), unlike before. therefore we have more "root files" than "file names".